### PR TITLE
Change from tabs selection to checkbox selection

### PR DESCRIPTION
**Pull request by alexbeno**

### CODE REVIEW

File: components/features/stacks/payments-flow/request-payments-stacks/features/views/select-rewards/components/selection-shortcut/selection-shortcut.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+}: TSelectionShortcut.Props) {
+  return (
+    <Flex justifyContent={"end"} alignItems={"center"} className="gap-2">
+      <Link.Button
+        className={cn(" text-spacePurple-500", {
+          "pointer-events-none !text-spaceBlue-300": !excludedRewards.length,
+        })}
+        onClick={onIncludeAll}
+      >
+        Select all
```
Comment (haydencleary) [inline components/features/stacks/payments-flow/request-payments-stacks/features/views/select-rewards/components/selection-shortcut/selection-shortcut.tsx:22]: Trad

File: components/features/stacks/payments-flow/request-payments-stacks/features/views/select-rewards/components/selection-shortcut/selection-shortcut.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      </Link.Button>
+      <Link.Button
+        className={cn("border-l border-l-card-border-light pl-2 text-spacePurple-500", {
+          "pointer-events-none !text-spaceBlue-300": !includedRewards.length,
+        })}
+        onClick={onExcludeAll}
+      >
+        Deselect all
```
Comment (haydencleary) [inline components/features/stacks/payments-flow/request-payments-stacks/features/views/select-rewards/components/selection-shortcut/selection-shortcut.tsx:30]: Trad

File: components/features/stacks/payments-flow/request-payments-stacks/features/views/select-rewards/components/selection-shortcut/selection-shortcut.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  excludedRewards,
+  includedRewards,
+  onExcludeAll,
+  onIncludeAll,
+}: TSelectionShortcut.Props) {
+  return (
+    <Flex justifyContent={"end"} alignItems={"center"} className="gap-2">
+      <Link.Button
+        className={cn(" text-spacePurple-500", {
+          "pointer-events-none !text-spaceBlue-300": !excludedRewards.length,
```
Comment (haydencleary) [inline components/features/stacks/payments-flow/request-payments-stacks/features/views/select-rewards/components/selection-shortcut/selection-shortcut.tsx:18]: Need to put a default cursor maybe ?

File: components/features/stacks/payments-flow/request-payments-stacks/features/views/select-rewards/components/selection-shortcut/selection-shortcut.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        })}
+        onClick={onIncludeAll}
+      >
+        Select all
+      </Link.Button>
+      <Link.Button
+        className={cn("border-l border-l-card-border-light pl-2 text-spacePurple-500", {
+          "pointer-events-none !text-spaceBlue-300": !includedRewards.length,
```
Comment (haydencleary) [inline components/features/stacks/payments-flow/request-payments-stacks/features/views/select-rewards/components/selection-shortcut/selection-shortcut.tsx:26]: Default cursor maybe ?

File: components/features/stacks/payments-flow/request-payments-stacks/features/views/select-rewards/select-rewards.tsx [hunk 5]
```diff
@@ -63,40 +66,9 @@ export function SelectRewards({
...
-            ))}
-          </div>
-        );
-      }
-
-      return null;
-    },
-    [excludedRewards, includedRewards]
-  );
+  function isRewardsInclude(rewardId: string) {
```
Comment (haydencleary) [inline components/features/stacks/payments-flow/request-payments-stacks/features/views/select-rewards/select-rewards.tsx:69]: memo ?

File: components/features/stacks/payments-flow/request-payments-stacks/features/views/select-rewards/select-rewards.tsx [hunk 6]
```diff
@@ -116,24 +88,48 @@ export function SelectRewards({
...
+                <div className="flex w-full flex-col py-6">
+                  <EmptyState
+                    illustrationSrc={IMAGES.global.categories}
+                    title={{ token: "v2.pages.stacks.request_payments.selectRewards.emptyState.title" }}
+                    description={{ token: "v2.pages.stacks.request_payments.selectRewards.emptyState.description" }}
+                  />
+                </div>
+              ) : (
+                <>
+                  {rewards.map(reward => {
```
Comment (haydencleary) [inline components/features/stacks/payments-flow/request-payments-stacks/features/views/select-rewards/select-rewards.tsx:117]: memo ?

**Review by haydencleary — CHANGES_REQUESTED** [review]

File: components/features/stacks/payments-flow/request-payments-stacks/features/views/select-rewards/components/selection-shortcut/selection-shortcut.tsx (hunk 3 above)
Comment (alexbeno) [reply components/features/stacks/payments-flow/request-payments-stacks/features/views/select-rewards/components/selection-shortcut/selection-shortcut.tsx:18]: don't got this one ? with we need a default ?

File: components/features/stacks/payments-flow/request-payments-stacks/features/views/select-rewards/select-rewards.tsx (hunk 5 above)
Comment (alexbeno) [reply components/features/stacks/payments-flow/request-payments-stacks/features/views/select-rewards/select-rewards.tsx:69]: callback ?

Comment (alexbeno) [reply components/features/stacks/payments-flow/request-payments-stacks/features/views/select-rewards/select-rewards.tsx:69]: this function is call in a map

**Review by haydencleary — APPROVED** [review]

**Pull request merged**